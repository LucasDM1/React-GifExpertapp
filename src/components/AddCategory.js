import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({setCategories}) => {

    const [inputValue, setinputValue] = useState("");

    const handleInputChange= (e) =>{

        setinputValue(e.target.value);

        console.log("handleInput ejecutado" )
    }

    const handleSubmit = (e) =>{
        e.preventDefault();
        console.log("submiteado")
        if (inputValue.trim().length>2) {
            setCategories(catg => [ inputValue, ...catg])
            setinputValue('')
        }

       
    }

    return (
        <form onSubmit={handleSubmit}>
            <p>{inputValue}</p>
            <input 
            type="text"
            value={inputValue}
            onChange={(e)=>{handleInputChange(e)}}
            />
        </form>
    )

   
}
 AddCategory.protoTypes = {
        setCategories: PropTypes.func.isRequired
    };