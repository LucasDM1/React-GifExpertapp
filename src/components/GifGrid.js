import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
//import { getGifs } from '../helpers/getGifs';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {

    // const [images, setImages] = useState([]);

    // useEffect(() => {
    //     getGifs(category)
    //         .then(setImages)
    // }, [category])
    const {loading, data} = useFetchGifs(category);
    

    return (
        <>
        <h2 className="animate__animated animate__fadeIn">{category}</h2>
        {loading && <p>Loading... </p>}
       
        <div className="card-grid">
            

            {data.map( img => (
                <GifGridItem 
                key={img.id}
                { ...img } 
                //img={img}
                />
            ))}
           
        </div>
        </>
    )
}

GifGrid.propTypes={
    category: PropTypes.string.isRequired

}