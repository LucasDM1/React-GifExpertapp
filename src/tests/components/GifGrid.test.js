import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
import '@testing-library/jest-dom';

jest.mock('../../hooks/useFetchGifs')
describe('Purebas en GifGrid', () => {
    
    const category='Hola';

    test('Debe mostrarse correctamente', () => {
        
        useFetchGifs.mockReturnValue({
            data:[],
            loading:true
        })
        const wrapper = shallow(<GifGrid category={category}/>);
        expect(wrapper).toMatchSnapshot();

    })

    test('Debe de mostrar items cuando se cargan imagenes useFetchGifs', () => {
        const gifs=[{
            
            id: 'ABC',
            title: 'Cualquier cosa',
            url: 'https://local/cualquier/cosa.jpg'
            
        }];
        
        useFetchGifs.mockReturnValue({
            data:gifs,
            loading:false
        })
    
        const wrapper = shallow(<GifGrid category={category}/>);   
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
        
    })
    
    
})
