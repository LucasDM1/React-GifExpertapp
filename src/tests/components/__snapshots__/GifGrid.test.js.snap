// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Purebas en GifGrid Debe de mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h2
    className="animate__animated animate__fadeIn"
  >
    Hola
  </h2>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://local/cualquier/cosa.jpg"
    />
  </div>
</Fragment>
`;

exports[`Purebas en GifGrid Debe mostrarse correctamente 1`] = `
<Fragment>
  <h2
    className="animate__animated animate__fadeIn"
  >
    Hola
  </h2>
  <p>
    Loading... 
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
